apps/base-docs/base-camp/docs/reading-and-displaying-data/useContractRead.md

```diff
@@ -0,0 +1,450 @@
+---
+title: The `userContractRead` Hook
+description: Learn how to call view and pure functions from a smart contract.
+hide_table_of_contents: false
+---
+
+The `useContractRead` hook is [wagmi]'s method of calling `pure` and `view` functions from your smart contracts. As with `useAccount`, `useContractRead` contains a number of helpful properties to enable you to manage displaying information to your users.
+
+---
+
+## Objectives
+
+By the end of this guide you should be able to:
+
+- Implement wagmi's `useContractRead` hook to fetch data from a smart contract
+- Convert data fetched from a smart contract to information displayed to the user
+- Identify the caveats of reading data from automatically-generated getters
+
+---
+
+## Contract Setup
+
+For this guide, we'll be continuing from the project you started for the [`useAccount` hook]. We'll work with an upgrade to the contract that you may have created if you completed the [ERC 20 Tokens Exercise]. See below for an example you can use if you don't already have your own!
+
+The contract creates a very simple DAO, in which users can create issues and vote for them, against them, or abstain. Anyone can `claim` 100 tokens. This is an insecure system for demonstration purposes, since it would be trivial to claim a large number of tokens with multiple wallets, then transfer them to a single address and use that to dominate voting.
+
+But it makes it much easier to test!
+
+:::caution
+
+If you're using your own contract, please redeploy it with the following `view` functions:
+
+```solidity
+function numberOfIssues() public view returns(uint) {
+    return issues.length;
+}
+
+function getAllIssues() public view returns(ReturnableIssue[] memory) {
+    ReturnableIssue[] memory allIssues = new ReturnableIssue[](issues.length);
+
+    for(uint i = 0; i < issues.length; i++) {
+        allIssues[i] = getIssue(i);
+    }
+
+    return allIssues;
+}
+```
+
+**You also need to make the `getIssue` function `public`. The original spec called for it to be `external`.**
```

Comment:
this may not matter a lot since we have a mix of things in the whole documentation but I think that in the most purist way, if we are not calling a function within the contract, should be external no?

Reply:
We're calling it internally with the new `getAllIssues` function I added.  So you can get one by id, or all.